lib/charms/layer/azure.py

```diff
@@ -174,7 +174,14 @@ def send_additional_metadata(request):
     available from the metadata server.
     """
     run_config = hookenv.config() or {}
-    res_grp = _azure("group", "show", "--name", request.resource_group)
+    res_grp = _azure(
+            "group",
+            "show",
+            "--subscription",
+            request.subscription_id,
+            "--name",
+            request.resource_group
+        )
```

Comment:
Perhaps handle the situation where the request doesn't yet have the `subscription_id` attribute.  Does the order of the args matter?  Otherwise this 'insert' in the middle is ridiculous

```suggestion
    res_grp_args = ["group", "show", "--name", request.resource_group]
    if hasattr(request, 'subscription_id'):
        for pos, arg in enumerate(["--subscription", request.subscription_id]):
            res_grp_args.insert(pos + 2, arg)

    res_grp = _azure(*res_grp_args)
```

Reply:
thnx, yes order doesnt matter will change this